client/services/FeedbackGeneratorService.js

```diff
@@ -373,21 +447,58 @@ tie.factory('FeedbackGeneratorService', [
        * @returns {Feedback}
        */
       getFeedback: function(tasks, codeEvalResult, rawCodeLineIndexes) {
+        // Reset all counters but consecutiveSameErrorCounter.
+        _resetCounters('consecutiveSameError');
+
+        // If the user recieves the same error message increment the same error
+        // counter.
+        if (previousErrorString === codeEvalResult.getErrorString()) {
+          consecutiveSameErrorCounter++;
+        }
+
         var feedback = _getFeedbackWithoutReinforcement(
           tasks, codeEvalResult, rawCodeLineIndexes);
         if (tasks.length > 0 && !codeEvalResult.getErrorString()) {
           feedback.setReinforcement(
             ReinforcementGeneratorService.getReinforcement(
               tasks[tasks.length - 1], codeEvalResult));
         }
+
+        // If the same error counter reaches the threshold, prompt the user to
+        // look at the primer.
+        if (consecutiveSameErrorCounter === UNFAMILIARITY_THRESHOLD) {
+          feedback.appendTextParagraph(_getUnfamiliarLanguageFeedback());
+          consecutiveSameErrorCounter = 0;
```

Comment:
Elena and I both agree with what you're saying, and without real data, it'll be hard to tell how accurate any of our language unfamiliarity triggers are. For now, our belief is that if someone is familiar with the language and its syntax, then they probably would have tried something logically different with their code after receiving the same error 2 or 3 times in a row, which would likely throw either other types of errors or at least different error tracebacks. After 5 times, it seems like the user might be grasping at straws. Nonetheless, we could also just leave this trigger out for now if we feel it's unnecessary.